src/certifier_algorithms.cc

```diff
@@ -49,15 +53,34 @@ const char * Enc_method_rsa_4096                  = "rsa-4096";
 const char * Enc_method_rsa_4096_private          = "rsa-4096-private";
 const char * Enc_method_rsa_4096_public           = "rsa-4096-public";
 const char * Enc_method_rsa_4096_sha384_pkcs_sign = "rsa-4096-sha384-pkcs-sign";
- 
+
+const char *Enc_public_key_algorithms[] =
```

Comment:
@rgerganov - This is existing code I added some many cycles ago. See PR #176.

I had tried to use C++ `String` in that work but could not do that for the reasons documented in [this comment](https://github.com/vmware-research/certifier-framework-for-confidential-computing/pull/176#discussion_r1287808897).

I don't think it's great huge value now to move to `std::vector` to refactor this plumbing.

If there is a strong need to do so, it will be better to refactor that in a different change-set once this one goes in. (But I won't have time to follow-up on that.)

Reply:
@rgerganov - On a 2nd read, I think I misunderstood your comment.

I guess, yes, for the new static string array I _could try_ to see if `std::vector` will work.

Let me get all the comments from @yelvmw and you, and will attempt this rework in one pass at the end, while applying changes to address all review feedback.

Reply:
@rgerganov - I amended the definitions of these string arrays to be like so:

```
 57 std::vector<const char *> Enc_public_key_algorithms =
 58                     {   Enc_method_rsa_2048
 59                       , Enc_method_rsa_3072
 60                       , Enc_method_rsa_4096
 61                       , Enc_method_ecc_384
 62                     };
 63
 64 const int Num_public_key_algorithms = Enc_public_key_algorithms.size();
 65
 66 // Names of Authenticated symmetric-key algorithms
 67 std::vector<const char *> Enc_authenticated_symmetric_key_algorithms =
 68                     {   Enc_method_aes_128_cbc_hmac_sha256
 69                       , Enc_method_aes_256_cbc_hmac_sha256
 70                       , Enc_method_aes_256_cbc_hmac_sha384
 71                       , Enc_method_aes_256_gcm
 72                     };
 73
 74 const int Num_symmetric_key_algorithms = Enc_authenticated_symmetric_key_algorithms.size();
 75
```

This change is the only item you requested.

So, as we discussed this morning, I am going to squash all commits and will merge this change-set into `/main` once CI passes.